### PR TITLE
Refactor metadata structure

### PR DESCRIPTION
**Pull request by jonnyshaw89**
Changes the allowed structure of metadata to 
```
"metadata": [
    {
        "name": "period_id",
        "validator": "string"
    },
```
from
```
"metadata": {
    "period_id": {
        "validator": "string"
    },
```

Json reformat is a pain, do we have any suggestions for a consistent formatter for JSON that we can automate?

### CODE REVIEW

File: schemas/questionnaire_v1.json [hunk 1]
```diff
@@ -62,30 +62,28 @@
       ]
     },
     "metadata": {
```
Comment (benhowes) [inline schemas/questionnaire_v1.json]: Is asserting the presence of `user_id` and `period_id` required as before? from a quick look at json schema, it looks like moving the checks to `_validate_schema_contain_metadata` is the only reasonable way to handle that check now.

**Comment by jonnyshaw89** [comment]
@benhowes comment addressed

File: app/validation/validator.py [hunk 2]
```diff
@@ -112,6 +112,12 @@ def _validate_schema_contain_metadata(self, schema):
         default_metadata = ['user_id', 'period_id']
         schema_metadata = [metadata_field["name"] for metadata_field in schema['metadata']]
 
+        if 'user_id' not in schema_metadata:
+            errors.append(self._error_message('Mandatory Metadata - user_id not specified in metadata field'))
+
+        if 'period_id' not in schema_metadata:
+            errors.append(self._error_message('Mandatory Metadata - period_id not specified in metadata field'))
```
Comment (benhowes) [inline app/validation/validator.py]: slightly pedantic, but, DRY. suggest:
```python
required_metadata_names = ('user_id', 'period_id')
for metadata_name in required_metadata_names:
    errors.append(self._error_message('Mandatory Metadata - `{}` not specified in metadata field'. format(metadata_name)))
```

Comment (jonnyshaw89) [reply app/validation/validator.py]: Fair, me being lazy

**Comment by jonnyshaw89** [comment]
@MebinAbraham all of "them"?

**Comment by MebinAbraham** [comment]
@jonnyshaw89 Ye?

File: tests/schemas/test_invalid_date_range_period.json [hunk 3]
```diff
@@ -1,66 +1,77 @@
...
+          "title": "Date Range Validation"
+        }
+      ],
+      "id": "default-section"
+    }
+  ],
+  "survey_id": "023",
+  "theme": "default",
+  "title": "Date formats"
 }
```
[the remark this answers is not in the document]
Comment (jonnyshaw89) [reply tests/schemas/test_invalid_date_range_period.json]: we don't have a `yarn format` on this repo?

[the remark this answers is not in the document]
Comment (insacuri) [reply tests/schemas/test_invalid_date_range_period.json]: Is that something worth adding?  It answers the question of automatically formatting JSON and seems to work pretty well in runner

[the remark this answers is not in the document]
Comment (jonnyshaw89) [reply tests/schemas/test_invalid_date_range_period.json]: Seems like a lot of overhead to add node as a dependency for this?

[the remark this answers is not in the document]
Comment (nealedj) [reply tests/schemas/test_invalid_date_range_period.json]: 👍 good suggestion though!

**Pull request merged**